ConfigurationSystem/Service/ConfigurationHandler.py

```diff
@@ -1,111 +1,111 @@
+
+class ConfigurationHandler(RequestHandler):
   """ The CS handler
   """
 
-  @classmethod
-  def initializeHandler( cls, _serviceInfo ):
-    """
-    Handler class initialization
-    """
-    # Check the flag for updating the pilot 3 JSON file
-    cls.updatePilotJSONFile = cls.srv_getCSOption( 'UpdatePilotCStoJSONFile', False )
-    if cls.updatePilotJSONFile:
-      cls.paramDict = {}
-      cls.paramDict['pilotFileServer'] = getServiceOption( _serviceInfo, "pilotFileServer", '' )
-      cls.paramDict['pilotRepo'] = getServiceOption( _serviceInfo, "pilotRepo", '' )
-      cls.paramDict['pilotVORepo'] = getServiceOption( _serviceInfo, "pilotVORepo", '' )
-      cls.paramDict['projectDir'] = getServiceOption( _serviceInfo, "projectDir", '' )
-      cls.paramDict['pilotVOScriptPath'] = getServiceOption( _serviceInfo, "pilotVOScriptPath", '' )
-      cls.paramDict['pilotScriptsPath'] = getServiceOption( _serviceInfo, "pilotScriptsPath", '' )
-
-    return S_OK( 'Initialization went well' )
-
   types_getVersion = []
-  def export_getVersion( self ):
-    return S_OK( gServiceInterface.getVersion() )
+
+  def export_getVersion(self):
+    return S_OK(gServiceInterface.getVersion())
 
   types_getCompressedData = []
-  def export_getCompressedData( self ):
+
+  def export_getCompressedData(self):
     sData = gServiceInterface.getCompressedConfigurationData()
-    return S_OK( sData )
+    return S_OK(sData)
+
+  types_getCompressedDataIfNewer = [basestring]
 
-  types_getCompressedDataIfNewer = [ basestring ]
-  def export_getCompressedDataIfNewer( self, sClientVersion ):
+  def export_getCompressedDataIfNewer(self, sClientVersion):
     sVersion = gServiceInterface.getVersion()
-    retDict = { 'newestVersion' : sVersion }
+    retDict = {'newestVersion': sVersion}
     if sClientVersion < sVersion:
-      retDict[ 'data' ] = gServiceInterface.getCompressedConfigurationData()
-    return S_OK( retDict )
+      retDict['data'] = gServiceInterface.getCompressedConfigurationData()
+    return S_OK(retDict)
 
-  types_publishSlaveServer = [ basestring ]
-  def export_publishSlaveServer( self, sURL ):
-    gServiceInterface.publishSlaveServer( sURL )
+  types_publishSlaveServer = [basestring]
+
+  def export_publishSlaveServer(self, sURL):
+    gServiceInterface.publishSlaveServer(sURL)
     return S_OK()
 
-  types_commitNewData = [ basestring ]
-  def export_commitNewData( self, sData ):
+  types_commitNewData = [basestring]
+
+  def export_commitNewData(self, sData):
     credDict = self.getRemoteCredentials()
-    if not 'DN' in credDict or not 'username' in credDict:
-      return S_ERROR( "You must be authenticated!" )
-    res = gServiceInterface.updateConfiguration( sData, credDict[ 'username' ] )
-
-    if self.updatePilotJSONFile:
-      if not res['OK']:
-        return res
-      return PilotCStoJSONSynchronizer( self.paramDict ).sync()
-    else:
+    if 'DN' in credDict or 'username' not in credDict:
```

Comment:
Comparing with the old version: It should also be `'DN' not in credDict`

Reply:
indeed! fixed that too